app/helpers/application_helper.rb

```diff
@@ -71,8 +71,18 @@ def copy_email_btn
             id: 'copy-filtered-speaker-emails'
   end
 
-  def bulk_finalize_button
-    link_to 'Bulk Finalize', bulk_finalize_event_staff_program_proposals_path, class: 'btn btn-warning navbar-btn'
+  def promote_button(program_session)
+    if program_session.can_promote?
+      link_to 'Promote', promote_event_staff_program_session_path(program_session.event, program_session), method: :patch, data: { confirm: "Are you sure you want to promote #{program_session.title}?" }, class: 'btn btn-warning'
+    end
+  end
+
+  def finalize_info
+    if current_event.proposals.soft_states.count > 0
```

Comment:
It's preferred to use .any?  vs count > 0.  I believe Rails may even optimize this at the AR level.